io/driver/module_unload_load.py

```diff
@@ -0,0 +1,140 @@
+    :param iteration: Number of time to unload and load the module
+    :only_io True for single provide module and False for All pci modules
+    """
+    def setUp(self):
+        """
+        get parameters.
+        """
+        self.module = self.params.get('module', default=None).split(',')
+        self.iteration = self.params.get('iteration', default=1)
+        self.only_io = self.params.get('only_io', default=None)
+        self.load_unload_sleep_time = 30
+        self.error_modules = []
+        self.mod_list = []
+        self.uname = linux_modules.platform.uname()[2]
+
+    def built_in_module(self, module):
+        """
+        checking whether the given module is built_in module or not
+        """
+        path = "/lib/modules/%s/modules.builtin" % self.uname
+        for each in genio.read_all_lines(path):
+            out = process.getoutput(each.split('/')[-1])
+            if module == out.split('.'[0]):
+                return True
+            return False
+
+    def get_depend_modules(self, module):
+        """
+        Returns the dependent modules
+        """
+        config_path = os.path.join(os.path.abspath(''),
+                                   "module_unload_load.py.data/config")
+        for line in genio.read_all_lines(config_path):
+            if module in line:
+                return line.split('=')[-1]
+
+    def flush_mpath(self, mdl):
+        """
+        flush the multipath
+        """
+        cmd = "lsmod | grep -i ^%s" % mdl
+        for i in range(20):
+            if process.getoutput(cmd).split(" ")[-1] == '0':
+                return True
+            process.system("multipath -F", ignore_status=True)
+            time.sleep(3)
+            continue
+        return False
+
+    def module_load_unload(self, module_list):
+        """
+        Unloading and loading the given module
+        """
+        for mod1 in module_list:
+            if linux_modules.module_is_loaded(mod1) is False:
+                linux_modules.load_module(mod1)
+                time.sleep(self.load_unload_sleep_time)
+
+        for _ in range(0, self.iteration):
+            for mdl in module_list:
+                sub_mod = self.get_depend_modules(mdl)
+                if sub_mod:
+                    for mod in sub_mod.split(' '):
+                        if mod == 'multipath':
+                            if self.flush_mpath(mdl) is False:
+                                self.error_modules.append(mdl)
+                            continue
+                        else:
+                            self.log.info("unloading sub module %s " % mod)
+                            process.system("rmmod %s" % mod)
+                self.log.info("unloading module %s " % mdl)
+                linux_modules.unload_module(mdl)
+                time.sleep(self.load_unload_sleep_time)
+                if linux_modules.module_is_loaded(mdl) is True:
+                    self.error_modules.append(mdl)
```

Comment:
Should we break here ?

Reply:
Can you clarify why you are doing that please ?

Reply:
this is right way, please have a close look.
taking each module from a list of modules and unload_load the modules along with thier sub-modules for those many iteration.

then go for next module.

Reply:
Or we can `continue`